src/main/webapp/script.js

```diff
@@ -143,18 +143,78 @@ function unfreezePage() {
  * Updates the page with new uploaded image.
  * It calls blurring method, puts new and blurred
  * images on canvases, updates blurRadius input
- * bar for new image and download button for
- * every new blurred image.
+ * bar for new image, download button for
+ * every new blurred image and adds click
+ * event on output canvas for specific blur.
  * @param {ImageObject} image
  */
 async function processImage(image) {
+  updateBlurRadiusInputBar(image);
+
+  updateBlurredImage(image);
+
   const outputCanvas = document.getElementById('output-canvas');
+  outputCanvas.onclick = (event) => {
+    toggleAreaBlur(event, image);
+  };
+}
 
-  updateBlurRadiusInputBar(image);
+/**
+ * Function to handle click on output canvas
+ * and toggle blurness of rects to blur
+ * if any were clicked.
+ * @param {MouseEvent} event
+ * @param {ImageObject} image
+ */
+function toggleAreaBlur(event, image) {
+  const outputCanvas = document.getElementById('output-canvas');
 
+  // coordinates of click.
+  const eventX = event.offsetX;
+  const eventY = event.offsetY;
+
+  let wasSomeRectClicked = false;
+
+  for (const blurRect of image.blurAreas) {
+    // Scale coordinate from image coordinates to canvas coordinates.
+    const scaleForCanvas = (coordinate) => {
+      return coordinate * outputCanvas.clientWidth / image.object.width;
+    };
+
+    // blurRect has all properties in original image coordinates,
+    // click event is in canvas coordinates.
+    // Scale blurRect to be in canvas coordinates.
+    const scaledRect = {
+      'leftX': scaleForCanvas(blurRect.leftX),
+      'topY': scaleForCanvas(blurRect.topY),
+      'rightX': scaleForCanvas(blurRect.leftX +
+          blurRect.width),
+      'bottomY': scaleForCanvas(blurRect.topY +
+          blurRect.height),
+    };
+
+    if (scaledRect.leftX <= eventX && eventX <= scaledRect.rightX &&
+        scaledRect.topY <= eventY && eventY <= scaledRect.bottomY) {
+      blurRect.toBeBlurred = !blurRect.toBeBlurred;
+      wasSomeRectClicked = true;
```

Comment:
You can make it more efficient with an early break - once you found the rect that was clicked there is no need to check other rects in the loop. You may refactor some pieces of this code into a separate function.

Reply:
I think you can click multiple rects at the same time, if they overlap, so it would be good to unblur both.

Reply:
I see, that's a fair point. Then it looks fine by me.